endpoints/events/event.go

```diff
@@ -185,15 +186,28 @@ func HandleAccountServiceErrors(errs []error) (status int, messages []string) {
 	messages = []string{}
 	status = http.StatusInternalServerError
 
+	//grab error messages
 	for _, er := range errs {
-		errCode := errortypes.ReadCode(er)
+		messages = append(messages, er.Error())
+	}
 
+	//set appropriate error code
+	for _, er := range errs {
+		if errors.Is(er, context.DeadlineExceeded) {
+			er = &errortypes.Timeout{
+				Message: er.Error(),
+			}
+		}
+
+		errCode := errortypes.ReadCode(er)
 		switch errCode {
 		case errortypes.BlacklistedAppErrorCode, errortypes.BlacklistedAcctErrorCode:
 			status = http.StatusServiceUnavailable
+			break
+		case errortypes.TimeoutErrorCode:
+			status = http.StatusGatewayTimeout
+			break
 		}
-
-		messages = append(messages, er.Error())
 	}
```

Comment:
I'm getting a bit of a "code smell" from the separate loops over the same data twice. I'm guessing the second loop allows you to break out early. I'm not worried about the performance. @bsardo what's your opinion here?

Reply:
I'm ok with the two loops assuming you're trying to break out early. However, I don't think the breaks are doing anything. If you're trying to break out of the for loop early, I think you need to label the for loop and then explicitly state that you're trying to break out of the for like this: `break myForLoopLabel`.

Reply:
Good point. That's breaking out of the switch case, which is implied anyway. Please don't use labels.

Reply:
🙂In that case, it's probably best to just iterate over the errors once without stopping early for a minor performance gain. You'd be setting the status to the last error instead of the first error but I imagine it's arbitrary either way.

Reply:
You're right about the `break`s, force of habit.. Refactored for the sake of simplicity.